In [3]:
import numpy as np
import pandas as pd

In [4]:
d={'ProductA':[25,30,28,32,26,29,27,33,31,28,30,29],'ProductB':[40,35,38,41,37,36,39,42,38,37,40,36]}

In [5]:
data=pd.DataFrame(d)
data

ProductA  ProductB
0         25        40
1         30        35
2         28        38
3         32        41
4         26        37
5         29        36
6         27        39
7         33        42
8         31        38
9         28        37
10        30        40
11        29        36

In [6]:
data.describe()


ProductA   ProductB
count  12.000000  12.000000
mean   29.000000  38.250000
std     2.374103   2.179449
min    25.000000  35.000000
25%    27.750000  36.750000
50%    29.000000  38.000000
75%    30.250000  40.000000
max    33.000000  42.000000

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   ProductA  12 non-null     int64
 1   ProductB  12 non-null     int64
dtypes: int64(2)
memory usage: 320.0 bytes


# 1.Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [5]:
data.mean()

ProductA    29.00
ProductB    38.25
dtype: float64

#### On average, ProductB has higher monthly sales compared to ProductA.

# 2.Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [7]:
data.median()

ProductA    29.0
ProductB    38.0
dtype: float64

* If the median are close to the mean, it suggests a relatively symmetric distribution of sales.
* If the median are significantly different from the mean, it indicates potential skewness in the sales distribution.

median is a measure of central tendency that is less sensitive to ex outliers than the mean. If there are outliers in the sales data, the median may provide a more robust representation of the central sales tendency.

# 3.Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [6]:
data.mode()

ProductA  ProductB
0      28.0        36
1      29.0        37
2      30.0        38
3       NaN        40

**Product A Mode:**

The mode represents the sales figure that occurs most frequently for Product A.
If there's more than one mode, it means that there are multiple sales values with the same highest frequency.


**Product B Mode:**

The mode represents the sales figure that occurs most frequently for Product B.
Similar to Product A, if there's more than one mode, it indicates multiple sales values with the same highest frequency.

# 4.Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.



In [8]:
weights = [1] * len(data)

# Calculate the weighted mean for Product A and Product B

weighted_mean_A = sum(data['ProductA'] * weights) / sum(weights)
weighted_mean_B = sum(data['ProductB'] * weights) / sum(weights)


In [9]:
weighted_mean_A 

29.0

In [10]:
weighted_mean_B

38.25

# Measures of Dispersion:


# 1.Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.


In [11]:
range_A = data['ProductA'].max() - data['ProductA'].min()
range_B = data['ProductB'].max() - data['ProductB'].min()

print(range_A)
print(range_B)

8
7


* larger range suggests more variability in sales, with potentially higher highs and lower lows.
* larger range indicates more variability in sales for Product B as well.

* If the range is large, it suggests that there are substantial differences in monthly sales, indicating higher variability.
*  smaller range implies more consistent monthly sales, indicating lower variability.

# 2.Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.


In [12]:
data.var()

ProductA    5.636364
ProductB    4.750000
dtype: float64

* More variability in sales values, indicating less predictability and potentially greater risk.


* Less spread in sales values, suggesting more consistent and predictable sales patterns.







# 3.Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.



In [13]:
data.std()

ProductA    2.374103
ProductB    2.179449
dtype: float64

* If the standard deviation is high, it means that monthly sales numbers vary a lot from the average. Some months may have significantly higher or lower sales than the typical month.

* If the standard deviation is low, it indicates that sales values tend to be close to the average. There's less variation from one month to another.

# Relative Measures of Dispersion:


# 1.Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.


In [14]:
mean_A = data['ProductA'].mean()
std_dev_A = data['ProductA'].std()
cv_A = (std_dev_A / mean_A) * 100

In [15]:
cv_A

8.186561038997239

The coefficient of variation is like a percentage that helps businesses compare how much a product's sales tend to vary compared to its average. If the percentage is higher, it means the sales are more unpredictable. If it's lower, the sales are more consistent. This measure is useful for understanding and comparing the risk associated with different products' sales patterns.

# 2.Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.


In [17]:
mean_B = data['ProductB'].mean()
range_B = data['ProductB'].max() - data['ProductB'].min()
relative_range_B = (range_B / mean_B) * 100


In [18]:
relative_range_B

18.30065359477124

The relative range accounts for the scale of sales data by expressing the range as a percentage of the mean. This normalization enables fair comparisons between products with different average sales, offering a proportional measure of variability relative to each product's average.







# 3.Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.



In [19]:
mean_A = data['ProductA'].mean()
std_dev_A = data['ProductA'].std()
rsd_A = (std_dev_A / mean_A) * 100


In [20]:
rsd_A

8.186561038997239

8.19% Relative Standard Deviation  suggests a moderate level of variability in the data relative to the mean. It implies that, on average, values deviate by approximately 8.19% from the mean, indicating a relatively stable pattern with some degree of fluctuation.